packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -17,200 +17,230 @@ import {
 
 describeCompat("GC trailing ops tests", "NoCompat", (getTestObjectProvider) => {
-	const tests = (tombstoneEnabled: boolean = false) => {
+	/**
+	 * @param transitionToReference - Whether the trailing op should transition objects to referenced or unreferenced.
+	 * @param beforeSweepTimeout - Whether the trailing op should be sent before or after sweep timeout.
+	 */
+	const tests = (transitionToReference: boolean, beforeSweepTimeout: boolean) => {
 		let provider: ITestObjectProvider;
-
-		const sweepTimeoutMs = 1;
-		const settings = {
-			"Fluid.GarbageCollection.ThrowOnTombstoneUsage": true,
-			"Fluid.GarbageCollection.TestOverride.SweepTimeoutMs": sweepTimeoutMs,
+		let settings = {};
+		let testContainerConfig: ITestContainerConfig;
+		let summarizerContainerConfig: ITestContainerConfig;
+
+		const sweepTimeoutMs = 100;
+		const gcOptions: IGCRuntimeOptions = {
+			inactiveTimeoutMs: 0,
+			enableGCSweep: true,
+			sweepGracePeriodMs: 0, // Skip Tombstone, these tests focus on Sweep
 		};
 
-		const gcOptions: IGCRuntimeOptions = { inactiveTimeoutMs: 0 };
-		const configProvider = tombstoneEnabled
-			? mockConfigProvider(settings)
-			: mockConfigProvider();
-		const tombstoneConfig: ITestContainerConfig = {
-			runtimeOptions: {
-				summaryOptions: {
-					summaryConfigOverrides: {
-						state: "disabled",
-					},
-				},
-				gcOptions,
-			},
-			loaderProps: { configProvider },
-		};
-		const testContainerConfig: ITestContainerConfig = tombstoneEnabled
-			? tombstoneConfig
-			: defaultGCConfig;
+		function updateDataStoreReferenceState(
+			fromDataStore: ITestDataObject,
+			dataStoreToUpdate: ITestDataObject,
+			key: string,
+			makeReferenced: boolean,
+		) {
+			if (makeReferenced) {
+				fromDataStore._root.set(key, dataStoreToUpdate.handle);
+			} else {
+				fromDataStore._root.delete(key);
+			}
+		}
 
 		/**
-		 * Submits a summary and returns the unreferenced timestamp for all the nodes in the container. If a node is
-		 * referenced, the unreferenced timestamp is undefined.
-		 * @returns a map of nodeId to its unreferenced timestamp.
+		 * Validates that the data store is not deleted from the data store and GC trees in the summary.
+		 * Also, it is referenced / unreferenced as expected.
 		 */
-		async function getUnreferencedTimestamps(summaryTree: ISummaryTree) {
+		function validateDataStoreStateInSummary(
+			summaryTree: ISummaryTree,
+			dataStoreNodePath: string,
+			expectGCStateHandle: boolean,
+			referenced: boolean,
+		) {
+			// Validate that the data store is not deleted from the data store summary tree.
+			const dataStoreId = dataStoreNodePath.split("/")[1];
+			const channelsTree = (summaryTree.tree[channelsTreeName] as ISummaryTree).tree;
+			assert.equal(
+				Object.keys(channelsTree).includes(dataStoreId),
+				true,
+				`Data store ${dataStoreId} should not have been deleted from the summary`,
+			);
+
+			// If expecting the GC state to be a handle, validate that and return.
```

Comment:
The converse isn't true?  I.e. it's not an "if-and-only-if" statement to say `true` for this param?

Reply:
Not sure what you mean. Do you mean if this is false, there should be validation that it's not a handle?

Reply:
Yeah, just asking

Reply:
It will be validated by the `getGCStateFromSummary` call below.